sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -128,30 +136,32 @@ private[sql] object PartitioningUtils {
   private[sql] def parsePartition(
       path: Path,
       defaultPartitionName: String,
-      typeInference: Boolean): Option[PartitionValues] = {
+      typeInference: Boolean): (Option[PartitionValues], Option[Path]) = {
```

Comment:
A base path is not always associated with a `PartitionValues`. If there is no partition, we can still have a base path.

That is why I don't make `case class PartitionValues(columnNames: Seq[String], literals: Seq[Literal])` to something like `case class PartitionValues(path: String, columnNames: Seq[String], literals: Seq[Literal])`.